sdk/azcore/tracing/tracing.go

```diff
@@ -189,3 +237,159 @@ type Attribute struct {
 	// Any other type will be formatted per rules of fmt.Sprintf("%v").
 	Value any
 }
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Link is the relationship between two Spans.
+type Link struct {
+	// SpanContext of the linked Span.
+	SpanContext SpanContext
+
+	// Attributes describe the aspects of the link.
+	Attributes []Attribute
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// SpanID is a unique identity of a span in a trace.
+type SpanID [8]byte
+
+// TraceID is a unique identity of a trace.
+type TraceID [16]byte
+
+// TraceFlags contains flags that can be set on a SpanContext.
+type TraceFlags byte
+
+// TraceStateImpl contains the implementation for TraceState.
+type TraceStateImpl struct {
+	// String contains the implementation for the TraceState.String method.
+	String func() string
```

Comment:
There are more methods to the Otel implementation for TraceState, but `String()` is all we need at the moment to parse it to use the [`ParseTraceState()`](https://pkg.go.dev/go.opentelemetry.io/otel/trace@v1.33.0#ParseTraceState) function

Reply:
Makes sense. Let's just start with the methods we know we need (we can always add more later).

Reply:
we don't  really need to modify or parse it - all we need is to propagate the string as is. I wonder it it's possible not to expose any of the TraceState-related features? We are able to propagate the context in Java and other languages without ever bothering with the tracestate.

Reply:
@karenychen is it possible to remove tracestate per @lmolkova comment?

Reply:
@lmolkova @jhendrixMSFT I changed `TraceState` into a simple string. However I don't think we can remove it from `SpanContextConfig` -- we will need it to convert between the wrapper `SpanContext` and the OTel `SpanContext`.

I cleaned up the azotel changes for us to see this better, but please do let me know if there is a better way we can do this: https://github.com/Azure/azure-sdk-for-go/blob/66f2c3bf70edf30fce250508b48a5a88570fca8a/sdk/tracing/azotel/otel.go#L141